### PR TITLE
Reduce number of per-PR CI jobs.

### PR DESCRIPTION
**Pull request by alliepiper**
#### Old

|#|Runner
|------|------
|327|`linux-amd64-cpu16`
|32|`linux-amd64-gpu-v100-latest-1`
|28|`linux-arm64-cpu16`
|15|`windows-amd64-cpu16`

#### New

|#|Runner
|------|------
|125|`linux-amd64-cpu16`
|25|`linux-amd64-gpu-v100-latest-1`
|15|`windows-amd64-cpu16`
|10|`linux-arm64-cpu16`

#### Old

|#|Runner
|------|------
|59|`linux-amd64-gpu-v100-latest-1`
|42|`linux-amd64-cpu16`
|12|`windows-amd64-cpu16`

#### New

|#|Runner
|------|------
|351|`linux-amd64-cpu16`
|59|`linux-amd64-gpu-v100-latest-1`
|28|`linux-arm64-cpu16`
|25|`windows-amd64-cpu16`

### CODE REVIEW

File: ci/matrix.yaml [hunk 1]
```diff
@@ -11,12 +11,11 @@ workflows:
...
-    - {jobs: ['build'], std: 'all', ctk: '11.1', cxx: ['gcc6', 'gcc7', 'gcc8', 'gcc9', 'clang9', 'msvc2017']}
-    - {jobs: ['build'], std: 'all', ctk: '11.8', cxx: ['gcc11'], sm: '60;70;80;90'}
+    - {jobs: ['build'], std: 'minmax', ctk: '11.1', cxx: ['gcc6', 'gcc9', 'clang9', 'msvc2017']}
     # Current CTK build-only
-    - {jobs: ['build'], std: 'all', cxx: ['gcc7', 'gcc8', 'gcc9', 'gcc10', 'gcc11', 'gcc12']}
-    - {jobs: ['build'], std: 'all', cxx: ['clang9', 'clang10', 'clang11', 'clang12', 'clang13', 'clang14', 'clang15', 'clang16', 'clang17']}
-    - {jobs: ['build'], std: 'all', cxx: ['intel', 'msvc2019']}
+    - {jobs: ['build'], std: 'minmax', cxx: ['gcc7', 'gcc8', 'gcc9', 'gcc10', 'gcc11', 'gcc12']}
+    - {jobs: ['build'], std: 'minmax', cxx: ['clang9', 'clang10', 'clang11', 'clang12', 'clang13', 'clang14', 'clang15', 'clang16', 'clang17']}
+    - {jobs: ['build'], std: 'minmax', cxx: ['intel', 'msvc2019']}
```
Comment (jrhemstad) [inline ci/matrix.yaml:18]: suggestion: What if we only test C++11/14 on the oldest version of each host compiler? As far as I see it, the purpose of C++11/14 testing is primarily to catch if we introduce using unguarded features that aren't supported in 11 or 14.

File: ci/matrix.yaml [hunk 2]
```diff
@@ -25,33 +24,33 @@ workflows:
     - {jobs: ['test_lid1',  'test_lid2'], project: ['cub'], std: 'max',    cxx: ['gcc']}
     - {jobs: ['test_nolid', 'test_lid0'], project: ['cub'], std: 'max',    cxx: ['clang', 'msvc']}
     # Modded builds:
-    - {jobs: ['build'], std: 'all', ctk: '12.5', cxx: 'nvhpc'}
-    - {jobs: ['build'], std: 'all', cxx: ['gcc', 'clang'], cpu: 'arm64'}
-    - {jobs: ['build'], std: 'all', cxx: ['gcc'], sm: '90a'}
+    - {jobs: ['build'], std: 'minmax', ctk: '12.5', cxx: 'nvhpc'}
+    - {jobs: ['build'], std: 'minmax', cxx: ['gcc', 'clang'], cpu: 'arm64'}
+    - {jobs: ['build'], std: 'minmax', cxx: ['gcc'], sm: '90a'}
```
Comment (jrhemstad) [inline ci/matrix.yaml:29]: Testing 17 and 20 is far more important to me on the latest version of compilers than testing 11 and 20.

File: ci/matrix.yaml (hunk 1 above)
Comment (alliepiper) [reply ci/matrix.yaml:18]: Done -- I also made the rest of these builds just do the highest supported standard to further reduce runner load.

**Comment by alliepiper** [comment]
I revised the matrix a bit more and addressed some feedback, the updated totals are in the PR description.

**Review by jrhemstad — APPROVED** [review]
I'd like @wmaxey to review as well.

**Review by wmaxey — APPROVED** [review]

**Pull request merged**